### DISCUSSION TITLE
Bug: ESLint doesn't find @typescript-eslint/eslint-plugin right after installing it

### DISCUSSION THREAD

**Discussion by beauxq** [eslint/eslint#15785] [General]
### Environment

Node version: 16.14.2
npm version: 8.5.0
Local ESLint version: 8.13.0
Global ESLint version: 8.13.0
Operating System: Ubuntu 20.04

### What parser are you using?

@typescript-eslint/parser

### What did you do?

```
$ npm install @typescript-eslint/eslint-plugin@latest --save-dev

up to date, audited 145 packages in 609ms

29 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
$ eslint .
```

### What did you expect to happen?

no error

### What actually happened?

```
Oops! Something went wrong! :(

ESLint: 8.13.0

ESLint couldn't find the plugin "@typescript-eslint/eslint-plugin".

(The package "@typescript-eslint/eslint-plugin" was not found when loaded as a Node module from the directory "/home/user".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install @typescript-eslint/eslint-plugin@latest --save-dev

The plugin "@typescript-eslint/eslint-plugin" was referenced from the config file in "PersonalConfig".

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.
```

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I'm trying to use the .eslintrc.json in my user home folder.

I install the requested package (both with and without `-g`) and it still keeps giving the same error.

**Comment by aladdin-add** [comment]
personal configuration has been deprecated since eslint v7: https://eslint.org/docs/user-guide/configuring/configuration-files#personal-configuration-files-deprecated

please use the [--config CLI option](https://eslint.org/docs/user-guide/command-line-interface#-c---config) instead.

**Comment by beauxq** [comment]
It's the same problem using `--config`

```
$ eslint . --config ~/.eslintrc.json

Oops! Something went wrong! :(

ESLint: 8.13.0

ESLint couldn't find the plugin "@typescript-eslint/eslint-plugin".

(The package "@typescript-eslint/eslint-plugin" was not found when loaded as a Node module from the directory "/home/user".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install @typescript-eslint/eslint-plugin@latest --save-dev

The plugin "@typescript-eslint/eslint-plugin" was referenced from the config file in "--config".

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.
```

**Comment by aladdin-add** [comment]
maybe need to install `@typescript-eslint/eslint-plugin` in `~/` (to be loadable with the config)?

**Comment by beauxq** [comment]
> maybe need to install @typescript-eslint/eslint-plugin in ~/ (to be loadable with the config)?

> I install the requested package (both with and without -g) and it still keeps giving the same error.

Isn't that what that does?

If not, how do I do that?

**Comment by aladdin-add** [comment]
try `cd ~ && npm i @typescript-eslint/eslint-plugin`

**Comment by beauxq** [comment]
> try `cd ~ && npm i @typescript-eslint/eslint-plugin`

My home directory has to be an npm package in order for it to work? That doesn't seem like a good solution.
Shouldn't it be able to use the global npm install? (`npm -g i @typescript-eslint/eslint-plugin`)

**Comment by aladdin-add** [comment]
just a workaround, and I agreed it's not ideal. In the early eslint, global-install usage was supported, but it's not well-designed and problematic.

another way is to use the option [--resolve-plugins-relative-to](https://eslint.org/docs/user-guide/command-line-interface#--resolve-plugins-relative-to).

**Comment by mrluisfer** [comment]
I run `npm list @typescript-eslint/eslint-plugin @typescript-eslint/parser` and it shows me some versions that are not well synchronized, maybe you could check if the installed version is the same that is being cached for your **.eslintrc**
```bash
yarn list v1.22.19
├─ @typescript-eslint/eslint-plugin@6.6.0
├─ @typescript-eslint/parser@6.6.0
└─ eslint-config-react-app@7.0.1
   ├─ @typescript-eslint/eslint-plugin@5.62.0
   └─ @typescript-eslint/parser@5.62.0
```

**Comment by costyn** [comment]
For anyone reading this in the future: I fixed this by installing/upgrading/migrating to eslint v9, using the migrate script to convert to a new config. Wasn't flawless but worked ok.

**Reply by ArthurRuxton-DY** [reply]
Im on v9, still not working

**Reply by maiconm** [reply]
I'm facing the same problem at v9. Could you solve it?

**Reply by ArthurRuxton-DY** [reply]
I can't remember but I believe this may have been to do with an incompatible version of node, check which version you have installed and which version is compatible with this version of eslint?

**Comment by sboagy** [comment]
Totally frustrating, I can't find a solution for this.  I just disabled LintLens for now.

**Comment by rimoslav** [comment]
The solution that worked for me was putting this setting inside my `.code-workspace` file:

```
"eslint.workingDirectories": [
	{
		"mode": "auto"
	}
]
```
